### ISSUE TITLE
infozip doesn't work.

### ISSUE THREAD

**Issue by cloudchou** [Shelnutt2/zip#1]
I clone your code and compile it. 
But when test on phones. it tell Segmentation fault.
Have you test ?

**Comment by Shelnutt2** [comment]
Yes and it is used in FreeGee today. What api level are you building for and what device/android version are you testing it on? I ended up compiling for ICS due to libc changes with newer ndks and wanting to maintain compatibility for ics and newer.

**Comment by cloudchou** [comment]
I build it on Cyanogenmod 10.1 for htc m7cdug device. 
But  linked staticly.
Can infozip support link staticly ?

**Comment by Shelnutt2** [comment]
I've only linked it dynamically. I've not tried static linking. Try adding -g to the CFLAGS and use the ndk debuger (gdb over adb)

**Comment by cloudchou** [comment]
I have tested linking dynamically. But it still say segementation fault.
![image](https://cloud.githubusercontent.com/assets/1670225/6518764/9fbd5a04-c3e5-11e4-865b-d2dca7847682.png)
I add some log in the code. finally I find that this function have problem:
zwchar *local_to_wide_string(local_string).
I will debug it with ndk debuger.

**Comment by cloudchou** [comment]
How do you debug ndk builed executable?
I got this error:
(gdb) target remote :5039
Remote debugging using :5039
Ignoring packet error, continuing...
warning: unrecognized item "timeout" in "qSupported" response
Ignoring packet error, continuing...
Ignoring packet error, continuing...
Ignoring packet error, continuing...
Ignoring packet error, continuing...
Ignoring packet error, continuing...
Ignoring packet error, continuing...
Malformed response to offset query, timeout

**Comment by cloudchou** [comment]
Now I find the reason. android ndk don't support unicode. So we have to remote -DUNICODE_SUPPORT'  from LOCAL_CFLAGS
